tests/test_examples.py

```diff
@@ -119,92 +121,90 @@ def generate_data(n_draws):
 
     assert np.max(np.abs(cdes_predict - cdes_coefs)) <= 1e-4
 
-
 def test_coef_predict_same_as_predict_xgb():
-  # Generate data p(z | x) = N(x, 1)
-  def generate_data(n_draws):
-    x = np.random.normal(0, 1, n_draws)
-    z = np.random.normal(x, 1, n_draws)
-    return x, z
+    # Generate data p(z | x) = N(x, 1)
+    def generate_data(n_draws):
+        x = np.random.normal(0, 1, n_draws)
+        z = np.random.normal(x, 1, n_draws)
+        return x, z
 
-  x_train, z_train = generate_data(10000)
-  x_validation, z_validation = generate_data(10000)
-  x_test, z_test = generate_data(10000)
+    x_train, z_train = generate_data(1000)
+    x_validation, z_validation = generate_data(1000)
+    x_test, z_test = generate_data(1000)
 
-  # Parameterize model
-  model = flexcode.FlexCodeModel(XGBoost, max_basis=31, basis_system="cosine",
-                                 regression_params={"max_depth": 5})
+    # Parameterize model
+    model = flexcode.FlexCodeModel(XGBoost, max_basis=31, basis_system="cosine",
+                                    regression_params={"max_depth": 5})
 
-  # Fit and tune model
-  model.fit(x_train, z_train)
-  model.tune(x_validation, z_validation,
-             bump_threshold_grid=np.linspace(0, 0.2, 3),
-             sharpen_grid=np.linspace(0.5, 1.5, 3))
+    # Fit and tune model
+    model.fit(x_train, z_train)
+    model.tune(x_validation, z_validation,
+                bump_threshold_grid=np.linspace(0, 0.2, 3),
+                sharpen_grid=np.linspace(0.5, 1.5, 3))
 
-  cdes_predict, z_grid = model.predict(x_test, n_grid=200)
+    cdes_predict, z_grid = model.predict(x_test, n_grid=200)
 
-  coefs = model.predict_coefs(x_test)
-  cdes_coefs = coefs.evaluate(z_grid)
+    coefs = model.predict_coefs(x_test)
+    cdes_coefs = coefs.evaluate(z_grid)
 
-  assert np.max(np.abs(cdes_predict - cdes_coefs)) <= 1e-4
+    assert np.max(np.abs(cdes_predict - cdes_coefs)) <= 1e-4
 
 
 def test_coef_predict_same_as_predict_lasso():
-  # Generate data p(z | x) = N(x, 1)
-  def generate_data(n_draws):
-    x = np.random.normal(0, 1, n_draws)
-    z = np.random.normal(x, 1, n_draws)
-    return x, z
+    # Generate data p(z | x) = N(x, 1)
+    def generate_data(n_draws):
+        x = np.random.normal(0, 1, n_draws)
+        z = np.random.normal(x, 1, n_draws)
+        return x, z
 
-  x_train, z_train = generate_data(10000)
-  x_validation, z_validation = generate_data(10000)
-  x_test, z_test = generate_data(10000)
+    x_train, z_train = generate_data(10000)
```

Comment:
These are still 10_000. Is this intended to be more of a load test?

Reply:
I dont think a load test is needed, but as I mentioned in the previous comment, the almost duplicate test in `test_cv_optim` exists because the regression classes uses two separate methods when the hyperparameters are just a single number vs a list of values. We either may think of consolidate the two methods or device a quicker test which checks both methods individually

Reply:
I've reduced the requested data size in the commit that I just pushed. The tests are now running in a little over 3 seconds on average now.